packages/confluence-adapter/src/filters/deploy_space_permissions.ts

```
@@ -0,0 +1,295 @@
+}
+
+const deployPermissions = async ({
+  spaceModificationChange,
+  ...args
+}: DeployPermissionsInput): Promise<DeployResult['errors']> => {
+  const changeData = getChangeData(spaceModificationChange)
+  const { permissionsToDeleted, permissionsToAdd } = calculatePermissionsDiff(spaceModificationChange)
+  log.debug('Found %o permissions to deleted and %o permissions to add', permissionsToDeleted, permissionsToAdd)
+  const permissionsToDeletedIds = permissionsToDeleted
+    .map(
+      permissionToDelete =>
+        changeData.value.permissionInternalIdMap?.[naclCase(createPermissionUniqueKey(permissionToDelete))],
+    )
+    .filter(values.isDefined)
+    .filter(_.isString)
+  const createPermissionRemoveChange = (id: string): RemovalChange<InstanceElement> => ({
+    action: 'remove',
+    data: {
+      before: new InstanceElement(id, permissionObjectType, { id }, [], {
+        [CORE_ANNOTATIONS.PARENT]: [changeData],
+      }),
+    },
+  })
+
+  const createPermissionAdditionChange = (permission: PermissionObject): AdditionChange<InstanceElement> => ({
+    action: 'add',
+    data: {
+      after: new InstanceElement(
+        createPermissionUniqueKey(permission),
+        permissionObjectType,
+        {
+          subject: {
+            type: permission.type,
+            identifier: permission.principalId,
+          },
+          operation: {
+            key: permission.key,
+            target: permission.targetType,
+          },
+        },
+        [],
+        { [CORE_ANNOTATIONS.PARENT]: [changeData] },
+      ),
+    },
+  })
+  const removalChanges = permissionsToDeletedIds.map(createPermissionRemoveChange)
+  const removalRes = await deployment.deployChanges({
+    changes: removalChanges,
+    ...args,
+  })
+  const additionChanges = permissionsToAdd.map(createPermissionAdditionChange)
+
+  const additionRes = await deployment.deployChanges({
+    changes: additionChanges,
+    ...args,
+  })
```

these can fail in various scenarios (if some API calls succeed and some fail), let's add a ticket to follow up with more safeties

specifically, what happens if we try to create a permission that already exists in the service? should we silently catch that specific error without failing?

---

https://salto-io.atlassian.net/browse/SALTO-5823